modules/core/site.js

```diff
@@ -280,6 +280,14 @@ var Hm_Ajax_Request = function() { return {
                 for (var name in res.folder_status) {
                     Hm_Folders.unread_counts[name] = res.folder_status[name]['unseen'];
                     Hm_Folders.update_unread_counts();
+                    const messages = new Hm_MessagesStore(name, Hm_Utils.get_url_page_number());
```

Comment:
Hello @jacob-js shouldn't we move the Hm_MessagesStore definition to core module ? Also on production I'm getting error
`site.js?v=somerandomchars:1282 Uncaught ReferenceError: Hm_MessagesStore is not defined`

Reply:
It’s not a good idea to move it to core. The core module is being misused, which is why you even find a reference to `Hm_MessagesStore` there. Core should not handle displaying the messages list or the message content; it should only contain things that can’t go into any other modules or are used by many, if not all, modules.

That’s my 2¢, unless I misunderstand your reasoning behind introducing the core module.

Reply:
The error in production should have been already addressed by https://github.com/cypht-org/cypht/pull/1264

Reply:
It really depends on the usage. Originally, we find message functions in core module as there are several modules that display messages - imap, the obsolete and removed pop3, jmap (which is working on top of imap currently), feeds (which are again treated as kind of messages, so Cypht can aggregate and show everyhting and other aggregate folders containing multiple sources). Thus, this message store makes sense to be in core module but I am not sure it was designed to work with anything outside imap module and maybe that's why Merci thinks it shouldn't be moved. What happens if you add an rss feed to Everything/All menu entry and browse that? Is the Hm_MessagesStore handling that fine?

Reply:
Hm_MessagesStore is called in core module. Once the imap module disabled it will raise an error as its definition won't be found. Also the places were it is used doesn't only concern the IMAP module

Reply:
Victor just made a very important point: `Hm_MessagesStore` works only with IMAP, so it makes sense for it to stay there. However, the usage found in core is incorrect, as it will break with other source types. This raises the question of whether we should extend its capabilities to work with all sources, requiring a move to the core module, or have a separate store for each source.

Reply:
It makes sense to extend it to work with other message types and move to core - since it handles the local storage cache, retrieving next items, etc. - these are actions relevant for all message types.

Reply:
This is happening in https://github.com/cypht-org/cypht/pull/1266